apps/src/code-studio/progressRedux.js

```diff
@@ -73,133 +76,149 @@ const initialState = {
  */
 export default function reducer(state = initialState, action) {
   if (action.type === INIT_PROGRESS) {
-    let stages = action.stages;
-    // Re-initializing with full set of stages shouldn't blow away currentStageId
-    const currentStageId =
-      state.currentStageId || (stages.length === 1 ? stages[0].id : undefined);
-    // extract fields we care about from action
     return {
       ...state,
       currentLevelId: action.currentLevelId,
       professionalLearningCourse: action.professionalLearningCourse,
       saveAnswersBeforeNavigation: action.saveAnswersBeforeNavigation,
-      stages: processedStages(stages, action.professionalLearningCourse),
-      lessonGroups: action.lessonGroups,
-      peerReviewLessonInfo: action.peerReviewLessonInfo,
+      lessonGroups: processLessonGroupData(
+        action.stages,
+        action.professionalLearningCourse,
+        action.lessonGroups,
```

Comment:
Here, if lessonGroups is not passed in the action, should the default value be used?